src/main/java/com/snowflake/kafka/connector/internal/OffsetContinuityRanges.java

```diff
@@ -0,0 +1,30 @@
+package com.snowflake.kafka.connector.internal;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.tuple.Pair;
+
+public class OffsetContinuityRanges {
+  private final List<Pair<Long, Long>> continuousOffsets;
```

Comment:
nit: we could also introduce a class "OffsetRange" with startOffsetInclusive, endOffsetExclusive but I'm fine with a pair

Reply:
Or a data class... oh no wait, it's not a Kotlin.
I think that Pair is good enough for this use case.